### PR TITLE
Feat/Auth forms CMS integration

### PR DESCRIPTION
**Pull request by m453h**
This PR integrates the Login, Sign Up, and Password Reset forms of the Civic Signal web tools with Payload CMS. 

We have introduced a [FormContentWrapper](https://github.com/CodeForAfrica/web-tools/blob/cd0c9b3499f43e5f061ff45ea207b578ce5d97a1/src/components/common/hocs/FormContentWrapper.js) that handles the data fetching for Form content from Payload CMS and made minimal changes to the `LoginForm.js`, `LoginFormContainer.js`,` RequestPasswordReset.js`, `SignUpContainer.js` and `ResetPasswordSuccessMessage.js`

Screenshots

Screenshot 2024-12-12 at 09 56 29
Screenshot 2024-12-13 at 08 45 42
Screenshot 2024-12-13 at 08 46 41

### CODE REVIEW

**Comment by kilemensi** [comment]
There are a lot of changes that we've already reviewed in https://github.com/CodeForAfrica/web-tools/pull/26. Can't we merge that one first?

File: src/components/user/SignupContainer.js [hunk 1]
```diff
@@ -42,15 +29,29 @@ class SignupContainer extends React.Component {
...
+      userPassword: { id: 'user.password.label', defaultMessage: content.passwordLabel },
+      userConfirmPassword: { id: 'user.confirm.password.label', defaultMessage: content.confirmPasswordLabel },
+      userNotes: { id: 'user.notes.label', defaultMessage: content.notesLabel },
+      notesHint: { id: 'notes.hint.label', defaultMessage: content.notesHint },
+      userSignup: { id: 'user.signup.label', defaultMessage: content.signUpButton },
+      userConsent: { id: 'user.consent.label', defaultMessage: serializeSlateToHtml(content.consentLabel) },
+
+    } : null;
+
+    return (localMessages ? (
```
Comment (kilemensi) [inline src/components/user/SignupContainer.js:48]: To reduce amount of changes, why not keep the name `messages` instead of `localMessages`?

**Review by kilemensi — APPROVED** [review]
Code LGTM!

File: src/components/user/SignupContainer.js (hunk 1 above)
Comment (m453h) [reply src/components/user/SignupContainer.js:48]: Sure I can go that 👍

**Pull request merged**